mdtraj/core/trajectory.py

```diff
@@ -1981,10 +1981,7 @@ def make_molecules_whole(self, inplace=False, sorted_bonds=None):
         if inplace:
             result = self
         else:
-            result = Trajectory(xyz=self.xyz, topology=self.topology,
-                                time=self.time,
-                                unitcell_lengths=self.unitcell_lengths,
-                                unitcell_angles=self.unitcell_angles)
+            result = self[:]
```

Comment:
You replace the `result` definition in a couple of spots with what is effectively an identical declaration (but not explicitly specifying unitcell conservation - perhaps would be good to add a comment as to what this achieves?

Reply:
Sure: the original definition results in attributes of the `result` object being views, not independent copies, of the original numpy arrays, so future alterations in the original trajectory object (e.g. changes in the coordinates) appear in the supposedly independent copy too. The revised `result = self[:]` syntax results in a proper deep copy and so fully independent objects.

Reply:
Sorry for the incredibly minor nitpick, but could you add a comment right above the `result = self[:]` noting that you've made a proper deep copy this way (and the importance)?  Just so we have it noted for future posterity/reduce future confusion? I can foresee this question coming up again in the future so figure we should head it off now.

Reply:
No problem - comment added.